pkg/deps/config/suppliers.go

```diff
@@ -77,14 +77,22 @@ func NewSupplyEventsQueryClientFn(queryNodeRPCURL *url.URL) SupplierFn {
 }
 
 // NewSupplyBlockClientFn supplies a depinject config with a blockClient.
-func NewSupplyBlockClientFn() SupplierFn {
+func NewSupplyBlockClientFn(queryNodeRPCURL *url.URL) SupplierFn {
 	return func(
 		ctx context.Context,
 		deps depinject.Config,
 		_ *cobra.Command,
 	) (depinject.Config, error) {
+
+		// Create a cosmos client from the queryNodeRPCURL used by the block client
+		// to initialize the block client by querying the latest block.
+		cosmosClient, err := cosmosclient.NewClientFromNode(queryNodeRPCURL.String())
```

Comment:
Ditto about naming of `cosmosClient` vs `cometClient` vs `cometBlockClient`. Pick one and be consistent